coil-base/src/main/java/coil/memory/RequestService.kt

```diff
@@ -145,10 +147,14 @@ internal class RequestService {
         return transformations.isEmpty() || Transformation.VALID_CONFIGS.contains(bitmapConfig)
     }
 
-    private fun Request.isConfigValidForHardware(size: Size): Boolean {
-        if (SDK_INT < O) return true
-        if (bitmapConfig != Bitmap.Config.HARDWARE) return true
-        return allowHardware && hardwareBitmapService.allowHardware(size)
+    private fun Request.isConfigValidForHardware(size: Size, target: Target?): Boolean {
+        if (SDK_INT >= O) {
+            if (bitmapConfig != Bitmap.Config.HARDWARE) {
+                return true
+            }
+        }
+
+        return !allowHardware || hardwareBitmapService.allowHardware(size, target)
```

Comment:
Can you keep the existing formatting when you add the extra check? Thanks!

Reply:
I can try :)